src/testing/flow_solver.h

```diff
@@ -33,9 +36,18 @@ class FlowSolver : public TestsBase
     /// Destructor
     ~FlowSolver() {};
 
+    /// Set up test
+    void setup_test() const;
+
     /// Runs the test (i.e. flow solver)
     int run_test () const;
 
+    /// Pointer to dg so it can be accessed externally. Mutable keyword necessary since functions are constant
+    mutable std::shared_ptr<DGBase<dim, double>> dg;
```

Comment:
If you want the flow_solver to hold a dg, create it at construction with an initializer list. mutable shared_ptr kind of entails we are OK with the shared_ptr to DG to change. You should be fine with creating an initial DG, keeping the shared_ptr constant, but modifying the underlying members of DG.

Reply:
Why does FlowSolver need to hold on to a DG and ODESolver?

Reply:
So that I can use FlowSolver within another class (fd_sensitivity) where I need two FlowSolvers and need to compare dg->solution at each time step. In the future, I might also want to use FlowSolver within other tests, and then I also may need access to dg->solution.
I think I may have a solution though. I am now trying to refactor FlowSolver so that it is composition-based instead of inheritance based. I will have an interface FlowSolverBaseCase, and FlowSolver will hold a FlowSolverBaseCase. This will also eliminate many of the virtualfunctions in FlowSolver.